test/Parse/statement_typo.swift

```diff
@@ -0,0 +1,36 @@
+// RUN: %target-typecheck-verify-swift
+
+// SR-8510
+
+func foo() -> Int? { return 0 }
+func bar() {
+  gaurd let x = foo() else { return }  // expected-error {{consecutive statements}} 
+  // expected-note@-1 {{did you misspell 'guard'}} {{3-8=guard}}
+  // expected-error@-2 {{use of unresolved identifier 'gaurd'}}
+  print(x)
+}
+
+func stmtMisspelledAtTheBegginingOfTheLineWithLet() {
+  gaurd let firstWithLet = foo() else { return } // expected-error {{consecutive statements}}
+  // expected-note@-1 {{did you misspell 'guard'}} {{3-8=guard}}
+  // expected-error@-2 {{use of unresolved identifier 'gaurd'}}
+  i let secondWithLet = foo() { return } // expected-error {{consecutive statements}}
```

Comment:
I suspect the previous recovery skips `i` here. Because, `use of unresolved identifier 'I'` is not emitted.
Could you split test cases to their own functions? like:
```
func stmtMisspelledAtTheBegginingOfTheLineWithLet1() {
   gaurd let firstWithLet = foo() else { return }
}
func stmtMisspelledAtTheBegginingOfTheLineWithLet2() {
   i let secondWithLet = foo() { return }
}
```

Reply:
Sure. Fixed.